changelog.md

```diff
@@ -1,10 +1,72 @@
+- Consistent command behavior including multi
+- Windows support
+- Improved performance
+- A lot of code cleanup
+- Many bug fixes
+- Better user support!
+
+## Features:
+
+- Added a "redis connection is broken" mode after reaching max connection attempts / exceeding connection timeout. (@BridgeAR)
+- Added NODE_DEBUG=redis env to activate the debug_mode (@BridgeAR)
+- Added a default connection timeout of 24h instead of never timing out as a default (@BridgeAR)
+- Errors thrown by redis will now include the redis error code as ".code" property. (@skeggse & @BridgeAR)
+- Errors thrown by node_redis will now include a ".command" property for the command used (@BridgeAR)
+- The entries in the keyspace of the server_info is now an object instead of a string. (@SinisterLight & @BridgeAR)
+- Small speedup here and there (e.g. by not using .toLowerCase() anymore) (@BridgeAR)
+- Full windows support (@bcoe)
+- Increased coverage by 10% and add a lot of tests to make sure everything works as it should (We reached 97% now) (@BridgeAR)
+- Remove dead code, clean up and refactor very old chunks (@BridgeAR)
+- Don't flush the offline queue if reconnecting (@BridgeAR)
+- Emit all errors insteaf of throwing sometimes and sometimes emitting them (@BridgeAR)
+- auth_pass passwords are now checked to be a valid password (@jcppman & @BridgeAR)
+- Support new commands and drop support for deprecated 'substr' (@BridgeAR)
+
+## Bug fixes:
+
+- Fix pubsub subscribe / unsubscribe taking down the app (@BridgeAR)
+- Don't kill the app anymore by randomly throwing errors sync instead of emitting them (@BridgeAR)
+- Don't catch user errors anymore occuring in callbacks (no try callback anymore & more fixes for the parser) (@BridgeAR)
+- Early garbage collection of queued items (@dohse)
+- Fix js parser returning errors as strings (@BridgeAR)
+- Do not wrap errors into other errors (@BridgeAR)
+- Authentication failures are now returned in the callback instead of being emitted (@BridgeAR)
+- Fix a memory leak on reconnect (@rahar)
+- Using `send_command` directly may no also be called without the args as stated in the [README.md](./README.md) @BridgeAR)
+- Fix the multi.exec error handling (@BridgeAR)
+- Fix commands being inconsistent and behaving wrong (@BridgeAR)
+- Channel names with spaces are now properly resubscribed after a reconnection (@pbihler)
+- Do not try to reconnect after the connection timeout has been exceeded (@BridgeAR)
+- Removed bad .eval implementation (@BridgeAR)
+- Fix commands not being rejected after calling .quit (@BridgeAR)
+- Fix .auth calling the callback twice if already connected (@BridgeAR)
+- Fix a couple small other bugs
+
+## Breaking changes:
+
+ ... connection is broken mode ...
+ ... multi error handling ...
+ ... not catching user errors anymore ...
+ ... emitting errors instead of throwing them sync ...
+ ... keyspace of the server_info
+ ... send_command commands have to be lowerCase
```

Comment:
might it be worth having a `.toLowerCase()` to protect legacy users, or would you rather just bite the bullet and force people to lower case commands?